tests/components/sunsynk/test_config_flow.py

```diff
@@ -0,0 +1,171 @@
+"""Test the Sunsynk config flow."""
+from unittest.mock import patch
+
+import pytest
+from sunsynk.client import InvalidCredentialsException
+from sunsynk.inverter import Inverter
+
+from homeassistant import config_entries
+from homeassistant.components.sunsynk.config_flow import (
+    CannotConnect,
+    InvalidAuth,
+    SunsynkHub,
+    validate_input,
+)
+from homeassistant.components.sunsynk.const import (
+    DATA_INVERTER_SN,
+    DATA_PASSWORD,
+    DATA_USERNAME,
+    DOMAIN,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+SUNSYNK_CLIENT_CREATE = "sunsynk.client.SunsynkClient.create"
+SUNSYNK_GET_INVERTERS = "sunsynk.client.SunsynkClient.get_inverters"
```

Comment:
Should not patch globally whenever possible. Patch the api instance within the ha component

Reply:
Fixed